src/instructlab/utils.py

```diff
@@ -215,18 +215,19 @@ def get_documents(
                 temp_dir=temp_dir,
                 skip_checkout=skip_checkout,
             )
-            file_contents = []
 
             logger.debug("Processing files...")
+            opened_files = False
             for pattern in file_patterns:
                 for file_path in glob.glob(os.path.join(repo.working_dir, pattern)):
                     if os.path.isfile(file_path) and file_path.endswith(".md"):
-                        with open(file_path, "r", encoding="utf-8") as file:
-                            file_contents.append(file.read())
+                        with open(file_path, "r", encoding="utf-8"):
```

Comment:
One thing we no longer will catch, by not reading the file, is any UTF-8 encoding problems in the file. This error has been observed in contributions.

Reply:
Which means that UTF-8 encoding issues would only be discovered when you run `ilab data generate` right?

I think that's fine. It's sort of surprising that we're doing any depth of validation in `ilab taxonomy diff` at all - given that it appears to just be a command that lists what files have changed

Reply:
`ilab taxonomy diff` is a combination of historically what were separate `list` and `check` commands that were combined a few months back, for context